### ISSUE TITLE
[Feature Request] Dynamically change the hotkey based on today's date

### ISSUE THREAD

**Issue by ma-sadeghi** [Vinzent03/obsidian-hotkeys-for-specific-files#5]
I know this might sound very user-specific, but I'll ask anyway.

So, I have my daily notes formatted by today's date as the title, ex. 2022-07-07, where I add my todo list for that specific dates plus some daily logs, which I want to keep for archive.

Anyway, the point is I go to my today's note quite often during the day, and I was wondering if I could create a hotkey that directly opens that up for me. The complication, of course, is that the title of today's date note changes on a daily basis. Not sure if that's possible to do using your plugin, but would really appreciate it.

**Comment by Vinzent03** [comment]
I think there is a default Obsidian command to open today's daily note. If so you can set a hotkey for that command directly.

**Comment by SimplGy** [comment]
I actually had this need and built my own plugin to support [opening a file by a customizable date pattern](https://github.com/SimplGy/obsidian-open-file-by-magic-date).

@ma-sadeghi , feel free to try it and report back.

@Vinzent03, would you want me to create a PR to "upstream" this capability into your plugin (and deprecate my own plugin)? The integration is probably very lightweight, it just looks for date syntax and applies the `moment` library if it finds it.

LOC: https://github.com/SimplGy/obsidian-open-file-by-magic-date/blob/master/src/main.ts#L171-L193

**Comment by Vinzent03** [comment]
@SimplGy Just to understand your plugin correctly. Despite the feature of being able to open a previous weekday, it does nothing different from using the inbuilt command to open today's daily note, right? And it's possible to run it on an additional directory to daily notes?

**Comment by SimplGy** [comment]
I would summarize the capability as:

* You can open a file defined by any of [Moment's allowed formatting options](https://momentjs.com/docs/#/displaying/format/).
  * eg: `{YYYY-MM}.md`, if you used a monthly notes file. Or `{YYYY-MM-DD h:mm:ss}.md` if you like your notes granular and very specifically time stamped.
* You can (optionally) shift the start day from "now" to any ISO weekday you want
  * eg: the previous monday: `Week of {mon:YYYY-MM-DD}.md`, or the previous saturday, etc.

The gap from existing plugins and capabilities, as I understand it, is:

1. weekly-notes only starts on Sunday (this is what started me on this customization journey)
2. daily notes is only for daily notes, of course :)

**Comment by ma-sadeghi** [comment]
@SimplGy Thank you so much for your suggestion, I tried the plugin, and I loved it. My daily journal is formatted as `Journal/2022/08 August/2022-08-22.md`, so I used `Journal/{YYYY/MM MMMM/YYYY-MM-DD}.md`, and it worked like a charm!

The only thing I miss is that the Daily Notes plugin can optionally insert a template upon note creation, but I can live with that since I only need to manually insert the template once. Having said that, if you think this feature is not entirely outside the scope of your plugin, I'd appreciate definitely appreciate it :grinning:
